File: backend/app/schemas/sqlldr.py
from pydantic import BaseModel
from typing import Optional


class SqlldrRequest(BaseModel):
    oracle_version: str = "19c"
    # Control file settings
    table_name: str = ""
    data_file: str = ""
    load_method: str = "insert"  # insert / append / replace / truncate
    fields_terminated_by: str = ","
    fields_optionally_enclosed_by: str = '"'
    lines_terminated_by: str = "\n"
    skip_rows: int = 0
    # Column definitions
    columns: str = ""  # comma-separated column definitions
    # Performance
    direct_path: bool = False
    parallel: bool = False
    bindsize: str = ""
    rows: str = ""
    errors: str = ""
    discardmax: str = ""
    # Other
    log_file: str = ""
    bad_file: str = ""
    discard_file: str = ""
    # Connection
    host: str = "localhost"
    port: int = 1521
    username: str = ""
    password: str = ""
    service_name: str = ""
    sid: str = ""
    connect_type: str = "service"


class SqlldrResponse(BaseModel):
    command: str
    control_file: str
    script: str
    warnings: list[dict] = []
    recommendations: list[str] = []

File: frontend/src/api/sqlldr.ts
import api from './index'

export interface SqlldrParams {
  table_name: string
  data_file: string
  load_method: string
  fields_terminated_by: string
  fields_optionally_enclosed_by: string
  lines_terminated_by: string
  skip_rows: number
  columns: string
  direct_path: boolean
  parallel: boolean
  bindsize: string
  rows: string
  errors: string
  discardmax: string
  log_file: string
  bad_file: string
  discard_file: string
  host: string
  port: number
  username: string
  password: string
  service_name: string
  sid: string
  connect_type: string
  oracle_home: string
}

export interface SqlldrResult {
  command: string
  control_file: string
  script: string
  warnings: { level: string; field: string; message: string; suggestion?: string }[]
  recommendations: string[]
}

export function generateSqlldr(data: SqlldrParams) {
  return api.post<SqlldrResult>('/sqlldr/generate', data)
}
